hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -1223,6 +1273,81 @@ Set<String> getDeltaFiles(OmSnapshot fromSnapshot,
             toSnapshot.getSnapshotTableKey()));
   }
 
+  /**
+   * Resolve the VersionMeta of the ancestor snapshot (fromSnapshotId)
+   * that the given snapshot (toSnapshot) was built on.
+   * Traverses the snapshot chain backwards using prevSnapId.
+   *
+   * @param toSnapshot the target snapshot.
+   * @param fromSnapshotId UUID of the source snapshot.
+   * @return the resolved VersionMeta of the source that was used to build the target.
+   */
+  private OmSnapshotLocalData.VersionMeta resolveBaseVersionMeta(
+      OmSnapshotLocalData toSnapshot,
+      UUID fromSnapshotId) throws IOException {
+    OmMetadataManagerImpl metadataManager =
+        (OmMetadataManagerImpl) ozoneManager.getMetadataManager();
+    // Start walking back from the child snapshot
+    OmSnapshotLocalData child = toSnapshot;
+    int numSnapshotsTraversed = 0;
+    while (numSnapshotsTraversed <= maxSnapshotLimit && !child.getPreviousSnapshotId().equals(fromSnapshotId)) {
+      UUID parentId = child.getPreviousSnapshotId();
+      // Load the parent snapshot in the chain
+      child = getSnapshotLocalData(
+          getSnapshotInfo(ozoneManager,
+              metadataManager.getSnapshotChainManager(),
+              parentId));
+      numSnapshotsTraversed++;
+    }
+    if (numSnapshotsTraversed > maxSnapshotLimit) {
+      LOG.error("Exceeded the traversal limit of {} while finding the VersionMeta of the fromSnapshot : {}" +
+          " that the toSnapshot was built on.", maxSnapshotLimit, fromSnapshotId);
+      return null;
+    }
+    SnapshotInfo snapshotInfo =
+        getSnapshotInfo(ozoneManager, metadataManager.getSnapshotChainManager(), fromSnapshotId);
+    OmSnapshotLocalData fromSnapshot = getSnapshotLocalData(snapshotInfo);
+    // Get the version that the child was built from
+    OmSnapshotLocalData.VersionMeta childVersionMeta = child.getVersionSstFileInfos().get(child.getVersion());
+    int versionUsedWhenBuildingChild = childVersionMeta.getPreviousSnapshotVersion();
```

Comment:
```suggestion
     return resolvedChildVersionMeta;
```

Reply:
Yeah this looks is simpler but think it's the same but just getting calculated in loop now and doesn't need extra handlimg outside the loop.